### ISSUE TITLE
Boss can jump "behind" background layers

### ISSUE THREAD

**Issue by odecay** [fishfolks/punchy#242]
I did notice a bug that appears regardless of these changes if any player/enemy is allowed to walk up past a certain y height. This can be reproduced on previous commits as well by extending the max Y value and moving the player upwards.
They become covered by the parallax background layers. This currently happens with the boss if he moves near the top of the walkable stage and then performs the jump for his smash attack.

_Originally posted by @odecay in https://github.com/fishfolks/punchy/issues/241#issuecomment-1229762033_

**Comment by odecay** [comment]
![sorting_punchy](https://user-images.githubusercontent.com/1379590/187249446-7755fd42-7fa1-44fc-9a04-0ed136eb650b.png)

example with player, note the z value and background sprite covering player

**Comment by edgarssilva** [comment]
We could change the base z value of fighters as well as the parallax layers in the `ParallaxResource`. I thought we add some constants for that but seems not so we could create some.

**Comment by odecay** [comment]
Would it not just get overwritten by the ysort system? I tried adding an offset in ysort system but it didn't seem to help

**Comment by odecay** [comment]
I also tried offsetting the parallax layers in the level file but it didn't behave as expected

**Comment by edgarssilva** [comment]
> Would it not just get overwritten by the ysort system? I tried adding an offset in ysort system but it didn't seem to help

The y-sorting should only add to the existing z value.

**Comment by odecay** [comment]
That would be compoundingly adding the y to the z every frame though

**Comment by edgarssilva** [comment]
> That would be compoundingly adding the y to the z every frame though

Ho right :P in my mind I thought we were adding then subtracting before and after render but I never actually ended up implementing it like that. 
I'll try some stuff out tomorrow and then submit something.

**Comment by odecay** [comment]
Im going to just add a set value for now so we can get a release out. I tried and it works with 300. But going to leave this open so we can come back to it.

**Comment by odecay** [comment]
https://github.com/fishfolks/punchy/pull/244

**Comment by edgarssilva** [comment]
> #244

Actually, I think you should keep the `ysort.0` because it differs on the sprite height, but keep the 300.

**Comment by odecay** [comment]
> > #244
> 
> ~Actually, I think you should keep the `ysort.0` because it differs on the sprite height, but keep the 300.~ Never mind you kept it to 0., ok then I'll check it out tomorrow.

yeah we arent using the Ysort offset right now, all fighters are initialized with 0. offset, but I think it will come in handy if we get to virtual Z for jumping so I think it makes sense to leave it. I could keep the ysort in that calculation AND add the offset but I'm going to leave it the way it is with the intent of removing the new offset from there entirely or moving it to a const if we decide there is not a better way to deal with it.
